services/notification/notifiers/status/base.py

```diff
@@ -103,11 +106,10 @@ def flag_coverage_was_uploaded(self, comparison) -> bool:
             len(report_uploaded_flags.intersection(flags_included_in_status_check)) > 0
         )
 
-    @property
-    def repository_service(self):
+    def repository_service(self, commit: Commit):
         if not self._repository_service:
-            self._repository_service = get_repo_provider_service(
-                self.repository, installation_name_to_use=self.gh_installation_name
+            self._repository_service = get_repo_provider_service_for_specific_commit(
```

Comment:
I just want to confirm my understanding of the code: does this pull the commit specific app that was set at the beginning of the app? Specifically, `_possibly_pin_commit_to_github_app` sets the app to use, then this function will retrieve it if it exists.

Reply:
That's the idea, yes